src/main/java/com.company/heap/RunningMedian.java

```diff
@@ -0,0 +1,104 @@
+package com.company.heap;
+
+import java.util.Collections;
+import java.util.PriorityQueue;
+
+public class RunningMedian {
+
+    public int[] findRunningMedian(int[] arr) {
+        int[] result = new int[arr.length];
+        PriorityQueue<Integer> minHeap = new PriorityQueue<>();
+        PriorityQueue<Integer> maxHeap = new PriorityQueue<>(Collections.reverseOrder());
+        for (int i = 0; i < arr.length; i++) {
+            if (maxHeap.isEmpty() || arr[i] <= maxHeap.peek()) {
+                maxHeap.add(arr[i]);
+            } else {
+                minHeap.add(arr[i]);
+            }
+            if (minHeap.size() - maxHeap.size() > 1) {
+                maxHeap.add(minHeap.poll());
+            }
+            if (maxHeap.size() - minHeap.size() > 1) {
+                minHeap.add(maxHeap.poll());
+            }
+            if (maxHeap.size() == minHeap.size()) {
```

Comment:
You could combine > and = in single if condition ( >= )

Reply:
changed